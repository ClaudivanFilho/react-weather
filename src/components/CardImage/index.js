import React, { Component, PropTypes } from 'react';

import './index.scss';

export default class CardImage extends Component {
  
  constructor(props) {
    super(props);
  }

  render() {
    const {title, image, width, children, color} = this.props; 
    return (
      <div className={`card-image ${color ? 'bg-' + color: 'bg-grey'}`}>
        <h2>{title}</h2>
        <div className="col-sm-12">
          { image && <img src={image} width={width || 200}/> }
        </div>
        <div className="col-sm-12">
          <h5>
            {children}
          </h5>
        </div>
      </div>
    );
  }
}

