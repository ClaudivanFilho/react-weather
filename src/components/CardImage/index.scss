.card-image {
  text-align: center;
  border: 3px solid rgba(0,0,0,0.08);
  border-radius: 10px;
  padding: 10px;
}

.bg-grey {
  background: #ffffff; /* Old browsers */
  background: -moz-linear-gradient(top, #ffffff 0%, #f3f3f3 33%, #ededed 54%, #ededed 60%, #ffffff 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(top, #ffffff 0%,#f3f3f3 33%,#ededed 54%,#ededed 60%,#ffffff 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to bottom, #ffffff 0%,#f3f3f3 33%,#ededed 54%,#ededed 60%,#ffffff 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#ffffff', endColorstr='#ffffff',GradientType=0 );
}

.bg-blue {
  background: #f2f9fe; /* Old browsers */
  background: -moz-linear-gradient(top, #f2f9fe 0%, #d6f0fd 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(top, #f2f9fe 0%,#d6f0fd 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to bottom, #f2f9fe 0%,#d6f0fd 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#f2f9fe', endColorstr='#d6f0fd',GradientType=0 );
}

.bg-yellow {
  background: #fefcea; /* Old browsers */
  background: -moz-linear-gradient(top, #fefcea 0%, #fffad3 99%); /* FF3.6-15 */
  background: -webkit-linear-gradient(top, #fefcea 0%,#fffad3 99%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to bottom, #fefcea 0%,#fffad3 99%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#fefcea', endColorstr='#fffad3',GradientType=0 );
}

.card-image:hover {
  opacity: 0.6;
}