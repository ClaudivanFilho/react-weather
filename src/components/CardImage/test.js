import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import CardImage from './index';

configure({ adapter: new Adapter() });

test('Test CardImage Component - without text description', () => {
  const component = shallow(<CardImage />);
  expect(component.text()).toEqual('');
});

test('Test CardImage Component - with text description', () => {
  const component = shallow(<CardImage>Teste</CardImage>);
  expect(component.text()).toEqual('Teste');
});

test('Test CardImage Snapshot', () => {
  const component = renderer.create(
    <CardImage width={50} title="Wind" color="blue">
      1 m/sec
    </CardImage>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})