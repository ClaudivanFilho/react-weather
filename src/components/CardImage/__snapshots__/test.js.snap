// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardImage Snapshot 1`] = `
<div
  className="card-image bg-blue"
>
  <h2>
    Wind
  </h2>
  <div
    className="col-sm-12"
  />
  <div
    className="col-sm-12"
  >
    <h5>
      1 m/sec
    </h5>
  </div>
</div>
`;
